docs/aws/audit/iamcompliance/rules/user_account_access_keys_use_remediation.mdx

```diff
@@ -1,6 +1,244 @@
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent inactive user account access keys in AWS IAM using the AWS CLI, you can follow these steps:
+
+1. **List All IAM Users**:
+   First, list all IAM users to identify which users have access keys.
+   ```sh
+   aws iam list-users
+   ```
+
+2. **List Access Keys for Each User**:
+   For each user, list their access keys to check their status and last used date.
+   ```sh
+   aws iam list-access-keys --user-name <username>
+   ```
+
+3. **Monitor Access Key Usage**:
+   Regularly monitor the usage of access keys to identify inactive keys. You can use the following command to get the last used date of each access key.
+   ```sh
+   aws iam get-access-key-last-used --access-key-id <access-key-id>
+   ```
+
+4. **Automate Inactive Key Deactivation**:
+   Create a script to automate the deactivation of access keys that have not been used for a specified period. Here is a basic example in Python:
+   ```python
+   import boto3
+   from datetime import datetime, timedelta
+
+   # Initialize a session using Amazon IAM
+   iam = boto3.client('iam')
+
+   # Define the inactivity period (e.g., 90 days)
+   inactivity_period = timedelta(days=90)
+
+   # Get the list of all IAM users
+   users = iam.list_users()['Users']
+
+   for user in users:
+       user_name = user['UserName']
+       access_keys = iam.list_access_keys(UserName=user_name)['AccessKeyMetadata']
+       
+       for key in access_keys:
+           access_key_id = key['AccessKeyId']
+           last_used_response = iam.get_access_key_last_used(AccessKeyId=access_key_id)
+           last_used_date = last_used_response['AccessKeyLastUsed'].get('LastUsedDate')
+           
+           if last_used_date:
+               if datetime.now(last_used_date.tzinfo) - last_used_date > inactivity_period:
+                   # Deactivate the inactive access key
+                   iam.update_access_key(UserName=user_name, AccessKeyId=access_key_id, Status='Inactive')
+           else:
+               # If the key has never been used, consider it inactive
+               iam.update_access_key(UserName=user_name, AccessKeyId=access_key_id, Status='Inactive')
+   ```
+
+By following these steps, you can ensure that inactive access keys are identified and deactivated, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent inactive user account access keys in IAM using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   - Install the Boto3 library if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+   from datetime import datetime, timedelta
+
+   # Initialize a session using Amazon IAM
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>